src/support/ScrollBar.ts

```diff
@@ -51,38 +53,73 @@ const DEFAULT_KEY = 'scrollbar';
+ * @param relativeSize The relative size to compare with the abosolute size
+ * @param absoluteSize The absolute size to compare with the relative size
+ */
+function fromRelative(relative: number, relativeSize: number, absoluteSize: number): number {
+	return absoluteSize * (relativeSize ? relative / relativeSize : 0);
+}
+
+/**
+ * Convert an absolute number to a relative number
+ * @param absolute The absolute value to convert to a relative value
+ * @param relativeSize The realtive size to compare with the absolute size
+ * @param absoluteSize The absolute size to compare with the relative size
+ */
+function toRelative(absolute: number, relativeSize: number, absoluteSize: number): number {
+	return absoluteSize ? (absolute / absoluteSize) * relativeSize : 0;
+}
+
+/**
  * A class that provides a visualization of scrolling as well as emits events when the user interacts with
  * the scroll bar.  The properties of the scroll widget are relative, thereby not needing to translate from
  * the real DOM size of the scroll bar or scroll area it represents.
  */
 @theme(css)
 export default class ScrollBar extends ThemeableBase<ScrollBarProperties> {
-	private _domSize = 0;
-	private _dragging = false;
-	private _dragPosition: number;
 	private _visible = false;
 
-	private _getSize(): number {
-		return this.properties.size || this._domSize;
-	}
-
-	private _fromRelative = (relative: number): number => {
-		const size = this._getSize();
-		return this._domSize * (size ? relative / size : 0);
-	}
-
-	private _toRelative = (absolute: number): number => {
-		const size = this._getSize();
-		return this._domSize ? (absolute / this._domSize) * size : 0;
+	/**
+	 * Returns the height or width of the scroll bar, depending on if it is horizontal or not
+	 */
+	private _getDomSize(): number {
+		const { horizontal, key = DEFAULT_KEY } = this.properties;
+		// TODO: Remove string coersion when dojo/widget-core#721 is published
+		const { height, width } = this.meta(Dimensions).get(`${key}`).size;
+		return horizontal ? width : height;
 	}
 
-	private _onDomUpdate(element: HTMLElement, key: string) {
-		const { horizontal = false, key: widgetKey = DEFAULT_KEY } = this.properties;
-		if (key === widgetKey) {
-			this._domSize = horizontal ? element.clientWidth : element.clientHeight;
+	/**
+	 * Determine if the scroll bar node is being clicked and if so, call the scroll listener with the relative position
+	 * that is attempted to be navigated to.
+	 * @param evt The mouse event
+	 */
+	private _onclick(evt: MouseEvent) {
+		// TODO: Remove string coersion when dojo/widget-core#721 is published
+		if (this.meta(Matches).get(`${this.properties.key}` || DEFAULT_KEY, evt)) {
+			evt.preventDefault();
+			const domSize = this._getDomSize();
+			const {
+				horizontal,
+				position,
+				size = domSize,
+				sliderMin = 10,
+				sliderSize,
+				onScroll
+			} = this.properties;
+			const absoluteDelta = (horizontal ? evt.offsetX : evt.offsetY) -
```

Comment:
i don't follow why you need to go from relative delta to absolute and back to relative again here. Can't you just record the `start` position of the first `pointerdown` and scroll relative to there? I may be missing something.

Reply:
The events works in absolute... so in order to calculate the delta.  The scroll bar only deals with relative values, so at some point you have to take the other information you need to calculate the delta, and convert it into absolute.  So this is taking an absolute offset and subtracting that from a relative internal state.  At some point, to compare apples and oranges, you need oranges and oranges, or apples and apples.